cosmos/launchpad/construction_derive_test.go

```diff
@@ -50,3 +50,32 @@ func TestLaunchpad_ConstructionDerive(t *testing.T) {
 	})
 	require.Equal(t, ErrUnsupportedCurve, deriveErr)
 }
+
+func TestRandomeAddrPrefix(t *testing.T) {
+	data, err := hex.DecodeString("A2FEB642851ACE7464999E56C8DBFD67C0A145E9")
+	require.NoError(t, err)
+
+	properties := rosetta.NetworkProperties{
+		Blockchain: "TheBlockchain",
+		Network:    "TheNetwork",
+		AddrPrefix: "rand",
+		SupportedOperations: []string{
+			"Transfer",
+			"Reward",
+		},
+	}
+
+	// Check using non cosmos address
+	adapter := NewLaunchpad(sdk.NewClient(""), tendermint.NewClient(""), properties)
+	deriveResp, deriveErr := adapter.ConstructionDerive(context.Background(), &types.ConstructionDeriveRequest{
+		PublicKey: &types.PublicKey{
+			Bytes:     data,
+			CurveType: "secp256k1",
+		},
+	})
+
+	require.Nil(t, deriveErr)
```

Comment:
```suggestion
	require.NoError(t, deriveErr)
```

Reply:
rosetta errors don't satisy error interface

Reply:
Yeah, that is correct, it cannot be done by noerror.